src/web/demo.rkt

```diff
@@ -228,83 +242,33 @@
    #:pcontext (run-herbie-command-pcontext cmd)
    #:profile? (run-herbie-command-profile? cmd)
    #:timeline-disabled? (run-herbie-command-timeline-disabled? cmd)))
-  (hash-set! *completed-jobs* job-id result)
-  (after-job-work result job-id (run-herbie-command-seed cmd)))
-
-(define (after-improve result job-id seed)
- (define path (format "~a.~a" job-id *herbie-commit*))
- (when (*demo-output*)
-    ;; Output results
-    (make-directory (build-path (*demo-output*) path))
-    (for ([page (all-pages result)])
-      (call-with-output-file (build-path (*demo-output*) path page)
-        (λ (out) 
-          (with-handlers ([exn:fail? (page-error-handler result page out)])
-            (make-page page out result (*demo-output*) #f)))))
-    (update-report result path seed
-                    (build-path (*demo-output*) "results.json")
-                    (build-path (*demo-output*) "index.html"))))
-
-; A place holder helper function that should be used when no work needs to be
-; done on the finished result
-(define (default-after result job-id seed) empty)
+  (hash-set! *completed-jobs* job-id result))
 
 (define (run-job job-info)
- (define sema (work-sema job-info))
- (define job-id (work-id job-info))
+ (match-define (work job-id info sema) job-info)
  (define path (format "~a.~a" job-id *herbie-commit*))
  (cond ;; Check caches if job as already been completed
   [(hash-has-key? *completed-jobs* job-id)
-  (semaphore-post sema)]
+   (semaphore-post sema)]
```

Comment:
Sorry I'm trying to make better diff's this PR has kinda turned into a mess. Might delete and just make a new one with one commit.